include/libssh2.h

```diff
@@ -239,7 +239,7 @@ typedef off_t libssh2_struct_stat_size;
 #define LIBSSH2_DH_GEX_OPTGROUP     1536
 #define LIBSSH2_DH_GEX_MAXGROUP     2048
 
-#define LIBSSH2_DH_MAX_MODULUS_BITS 10000
+#define LIBSSH2_DH_MAX_MODULUS_BITS 16384
```

Comment:
It is pretty obvious from the diff what the change does.  It would be good if the commit message explained why the change was needed or useful.

Reply:
Sure, the value 10000 was pulled out of OpenSSH, which is the value I've been using for a while in my fork without issue. With this commit making into the official release, I double-checked OpenSSH and their max values change if using RSA or DSA keys so I bumped this value to the larger number. This is to prevent DOS with extremely large values being computed.